README.md

```diff
@@ -14,6 +14,54 @@ This project is based on [TinGen](https://github.com/eXhumer/TinGen) by [eXhumer
 - credentials.json (you can modify location & name with `--credentials` flag) It can be obtained from [here](https://developers.google.com/drive/api/v3/quickstart/python) by clicking the Enable Drive API button in there while being signed in with the user account you want to generate credentials for or from Google's Developer Console.
 - Google Drive Folder IDs to scan and index
 
+## Usage
+
+**NOTE:** the first time `rustfoil.exe` is ran, a URL will be displayed. Copy and paste that URL into your web browser and follow the instructions on screen to continue. This process generates a `token.json` file that allows rustfoil to access your drive.
+
+- To use rustfoil to generate an `index.tfl` file corresponding to your Google Drive folder, run the following command:
+
+```
+rustfoil.exe GOOGLE_DRIVE_FOLDER_ID
+```
+
+This will generate an `index.tfl` file in the same directory that `rustfoil.exe` is located.
+
+- For a more comprehensive guide, check out [ii0606226](https://github.com/ii0606226)'s step-by-step guide [here](https://ii0606226.github.io/gdrive-compendium/rustfoil-steps)!
+
+## Flags and Options
+
+Flag | Description
+:--- | :---
+`--add-non-nsw-files` | Adds files without valid NSW ROM extension(NSP/NSZ/XCI/XCZ) to index
+`--add-nsw-files-without-title-id` | Adds files without valid Title ID
+`-h`, `--help` | Prints help information
+`--headless` | If OAuth should be done headless
```

Comment:
Here you should probably explain what headless means, some people might not be aware of the term